### ISSUE TITLE
Phase out "Jump with Up" support

### ISSUE THREAD

**Issue by maxteufel** [SuperTux/supertux#499]
Phase out "Jump with Up" support by first adding a deprecation notice to the 0.5 release and then removing it entirely from the following releases.

It is already causing issues, and will be causing more issues when swimming is implemented for too little gain.

**Comment by brmbrmcar** [comment]
No, I hate this idea. I use it as I find it natural. If you want another key for other stuff, by all means do so. It is not necessary to remove Jump with Up.

**Comment by maxteufel** [comment]
You should be able to set Jump to Up anyway and change Up to something else.

**Comment by tobbi** [comment]
@brmbrmcar You hate everything. I'd hardly count your opinion at this point.

**Comment by brmbrmcar** [comment]
@maxteufel That would be annoying. It would just be easier to add another key. As for the problems, they are easy to solve.

And if you won't accept an opinion you shouldn't do this kind of thing.

**Comment by tobbi** [comment]
You've been criticizing every *single* thing.

**Comment by maxteufel** [comment]
> That would be annoying.

Why? `/* Insert arguments here */`

> It would just be easier to add another key. As for the problems, they are easy to solve.

Please don't say things when you don't actually know what you're talking about. In fact, this is *not* easy to solve. At least without many workarounds. Which makes the feature worthless because it's too much work to maintain it. Here's a discussion from IRC.

> 2016-07-17 15:15:39 \<@maxteufel\> The easiest solution, making the feature inavailable only when it can't be used (e.g. door, swimming), would cause inconsistencies
> 2016-07-17 15:15:58 \<@Karkus476\> @maxteufel: And lots of confusion.
> 2016-07-17 15:16:00 \<@maxteufel\> The other solution would be to have a power-up key, and an action key.
> 2016-07-17 15:16:40 \<@maxteufel\> But this is a direction thing in this case [swimming], an action key would be of no use

**Comment by brmbrmcar** [comment]
Why not disable Jump with Up, but allow users to allocate the same key to both with a warning?

**Comment by maxteufel** [comment]
The idea to make a second action key wouldn't make sense too. Here's a discussion from IRC.

> 2016-07-17 15:15:39 \<@maxteufel\> The easiest solution, making the feature inavailable only when it can't be used (e.g. door, swimming), would cause inconsistencies
> 2016-07-17 15:15:58 \<@Karkus476\> @maxteufel: And lots of confusion.
> 2016-07-17 15:16:00 \<@maxteufel\> The other solution would be to have a power-up key, and an action key.
> 2016-07-17 15:16:40 \<@maxteufel\> But this is a direction thing in this case [swimming], an action key would be of no use

**Comment by maxteufel** [comment]
If we want to start phasing this out in 0.5.0, we have to make a decision on this **now**!

I'm in favor of starting this deprecation phase. Adding a notice to the 0.5.0 release won't hurt anything, and won't affect the feature for now (if we decided to reintroduce it/undeprecate it, that's still possible).

**Comment by maxteufel** [comment]
As of commit 4b9ae3988d4896fdfb9f353cd65a092b017060fb, this has been marked as deprecated!
